src/include/miopen/rnn_util.hpp

```diff
@@ -35,6 +35,55 @@
 
 namespace miopen {
 
+enum class RnnDirection
+{
+    Forward  = 0,
+    Backward = 1
+};
+
+struct RnnBatches
```

Comment:
```
struct X{
 int at(int time, RnnDirection direction){
     return RnnDirection::Forward ? batches[time] : batches[ (batches.size() - 1) - time];
 }
...
 std::vector<int> batches;
}
```
this is true for any direction 
`.next(time, direction) == .at(time+1, direction)`
`.prev(time, direction) == .at(time-1, direction)`

`next_time()` and `prev_time()` are redundant functions.

<details>
  <summary>explanation</summary>

  FWD next_time:
  `cur_time(time, direction) + 1` => **`(time+1)`**
  
  BWD next_time:
  `cur_time(time, direction) - 1` => `(batches.size() - time - 1) - 1` => `(batches.size() - `**`(time+1)`**` - 1)`

  BWD prev_time:
  `cur_time(time, direction) + 1` => `(batches.size() - time - 1) + 1` => `(batches.size() - `**`(time-1)`**` - 1)`

</details>

`next()` and `prev()` can be used as syntactic sugar if you want, but than remove mix using of `+1` and `next()` in code